src/components/coin.ts

```diff
@@ -135,6 +140,21 @@ export const changeDbCoinBalanceByUserId = async (
   await createCoinLedgerEntry(userId, oldBalance, newBalance, event, reason, adminId);
 };
 
+/*
+  Get the leaderboard for the current coin amounts.
+*/
+export const getCurrentCoinLeaderboard = async (): Promise<UserCoinEntry[]> => {
+  const db = await openDB();
+  const res = await db.all(
+    `
+      SELECT user_id, balance
+      FROM user_coin
+      ORDER BY balance DESC
```

Comment:
limit by same const

Reply:
We need to get all because we get the current position of the user as well

Reply:
oh, hmmmmmm... maybe a count of all-greater-than with a lookup of this user's coin would be the best performance thingy

Reply:
Yeah I think that would be good, I'll implement that change